core/src/main/scala/com/colisweb/jrubysnesshours/core/Schedule.scala

```diff
@@ -12,18 +15,42 @@ final case class Schedule private[core] (
     exceptions: Map[LocalDate, List[TimeInterval]],
     timeZone: ZoneId
 ) {
+
+  def splitTimeSegmentsSingleDate(date: LocalDate, hours: Long): List[TimeInterval] =
+    splitTimeSegments(zoned(date.atTime(MIN)), zoned(date.atTime(MAX)), hours).map(_.interval)
+
+  def splitTimeSegments(
+      start: ZonedDateTime,
+      end: ZonedDateTime,
+      hours: Long,
+      cutOff: Option[DoubleCutOff] = None
+  ): List[TimeIntervalForDate] = {
+    val startTime = local(start).toLocalTime
+
+    for {
+      nextWorkingDay <- nextOpenTimeAfter(start.plusDays(1).withHour(0).withMinute(0)).toList
+      localStart = cutOff.fold(start.toLocalDateTime)(
+        _.nextAvailableMoment(startTime, start.toLocalDate, nextWorkingDay.toLocalDate)
+      )
+      interval <- intervalsBetween(localStart, local(end))
```

Comment:
Quelque chose m'échappe dans cette méthode.. 

Au vus de sont utilisation dans la methode `splitTimeSegmentsSingleDate` ( visiblement uniquement utilisée dans les tests ??? ), on est d'accord qu'on appel `splitTimeSegments` toujours pour un seul jour ( start = beginning of day d'un jour, end = end of day du même jour ). 

la variable `localStart` utilise le résultat de `nextAvailableMoment`, et si on passe dans le dernier `else` de `nextAvailableMoment`, on renvois `nextDate.atTime(nextDay.firstAvailableTime)`, et donc une LocalDateTime qui n'est pas le même jour que notre `start` ou `end`.

Imaginons maintenant qu'on ai le cas suivant : `start` = "2019-01-01 00h00" et `end` = "2019-01-01 23h59" et que notre `nextAvailableMoment` nous donne un résultat pour le jours suivant, disons "2019-01-02 10h00" , alors on va appeler juste en dessous : 
`intervalsBetween("2019-01-02 10h00", 2019-01-01 23h59")` ce qui va faire planter l'assertion :/ 

Ou alors j'ai loupé une étape.. mais si ce que je dis est vrais, on a un bug

Reply:
D'ailleurs je viens en fait de voir que la methode `intervalsBetween` n'as pas d'assertion, on peux donner un startTime > au endTime, je suis pas sur que ce soit bon .. 🤔